haste_storage_client/core.py

```diff
@@ -0,0 +1,46 @@
+from pymongo import MongoClient
+from keystoneauth1 import session
+import swiftclient.client
+
+
+class HasteStorageClient:
+    def __init__(self,
+                 stream_id,
+                 host,
+                 port,
+                 keystone_auth):
+        """
+        :param stream_id: String ID for the stream session - used to group all the data
+        (unique for each execution of the experiment)
+        :param host: Hostname/IP of database server.
+        :param port: Database server port. Usually 27017.
+        :param keystone_auth: OpenCloud keystone auth v3 password object,
+        see: https://docs.openstack.org/keystoneauth/latest/api/keystoneauth1.identity.v3.html#module-keystoneauth1.identity.v3.password
+        """
+        self.mongo_client = MongoClient(host, port)
+        self.mongo_db = self.mongo_client.metadata
+        self.stream_id = stream_id
+
+        keystone_session = session.Session(auth=keystone_auth)
+        self.swift_conn = swiftclient.client.Connection(session=keystone_session)
```

Comment:
Should the connection open at construction or the first time it is used?

Reply:
I did it this (a) its more likely to "fail fast" - catch the errors sooner (b) don't have the complexity of tracking the 'first time' in the save and close - its simpler.

I guess also, this way we also "inherit" any thread safety of the underlying clients viz. the save() method. (otherwise our 'first time' check must also be thread safe.)

Reply:
I see. Sounds good.